notebooks/rocoto.ipynb

```diff
@@ -0,0 +1,1060 @@
+   "cell_type": "markdown",
+   "id": "40e85303-be55-4ebc-8f78-d9a0373cbf09",
+   "metadata": {},
+   "source": [
+    "## Validating Workflows\n",
+    "\n",
+    "The `rocoto.validate()` function checks Rocoto XML files against its schema to check for any errors.\n",
+    "<!--cell 38-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "6a681e6d-800c-4d9f-87a0-270e72dcb7be",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Help on function validate in module uwtools.api.rocoto:\n",
+      "\n",
+      "validate(xml_file: Union[str, pathlib.Path, NoneType] = None, stdin_ok: bool = False) -> bool\n",
+      "    Validate purported Rocoto XML file against its schema.\n",
+      "\n",
+      "    :param xml_file: Path to XML file (``None`` or unspecified => read ``stdin``).\n",
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: ``True`` if the XML conforms to the schema, ``False`` otherwise.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(rocoto.validate)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "91f72e92-083c-4c36-90d8-1860133fe33b",
+   "metadata": {},
+   "source": [
+    "The following is a minimal Rocoto workflow which contains identical tags, attributes, and values as the simple workflow example from the [Building Rocoto Workflows with UW YAML](#Building-Rocoto-Workflows-with-UW-YAML) section above.\n",
+    "<!--cell 40-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "22a8fe77-2094-4139-9ff2-91dc897c3af3",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<?xml version='1.0' encoding='utf-8'?>\n",
+      "<workflow realtime=\"False\" scheduler=\"slurm\">\n",
+      "  <cycledef>202410290000 202410300000 06:00:00</cycledef>\n",
+      "  <log>logs/test.log</log>\n",
+      "  <task name=\"greet\">\n",
+      "    <cores>1</cores>\n",
+      "    <walltime>00:00:10</walltime>\n",
+      "    <command>echo Hello, World!</command>\n",
+      "    <jobname>greet</jobname>\n",
+      "  </task>\n",
+      "</workflow>\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/rocoto/simple-workflow.xml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb04cf78-1afd-48eb-8673-803fbabac836",
+   "metadata": {},
+   "source": [
+    "`validate()` accepts <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> objects or string paths passed to the `xml_file` parameter. It's also possible to set `xml_file` to `None` and read the file from `stdin` if `stdin_ok` is set to `True`, but this is a rare use case that won't be covered here. The function returns `True` if the XML file is validated without any errors, and `False` otherwise. The number of schema-validation errors, as well as details on the errors (if any), are reported.\n",
```

Comment:
```suggestion
    "`validate()` accepts <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> objects or string paths passed via the `xml_file` parameter. (If `xml_file` is omitted or `None`, and `stdin_ok` is `True`, XML will be read from `stdin`, but this is a rare use case that won't be covered here.) The function returns `True` if the XML is validated without any errors, and `False` otherwise. The number of schema-validation errors, as well as details on the errors (if any), are reported.\n",
```